jwst/background/background_step.py

```diff
@@ -58,13 +57,37 @@ def process(self, input, bkg_list):
                 result = background_sub.subtract_wfss_bkg(
                                 input_model, bkg_name, wlrange_name)
                 result.meta.cal_step.back_sub = 'COMPLETE'
-
             else:
+                # check if input data is NRS_IFU
+                tolerance = 1.0e-8
+                do_sub = True
+                if input_model.meta.instrument.name in ["NIRSPEC"]:
+                    # check if GWA_XTIL & GWA_YTIL values of source
+                    # background are the same. If not skip step
+                    input_xtilt = input_model.meta.instrument.gwa_xtilt
+                    input_ytilt = input_model.meta.instrument.gwa_ytilt
+                    for bkg_file in bkg_list:
+                        bkg_model = datamodels.open(bkg_file)
+                        bkg_xtilt = bkg_model.meta.instrument.gwa_xtilt
+                        bkg_ytilt = bkg_model.meta.instrument.gwa_ytilt
+                        xdiff = abs(bkg_xtilt - input_xtilt)
+                        ydiff = abs(bkg_ytilt - input_ytilt)
+                        bkg_model.close()
+                        if xdiff > tolerance or ydiff > tolerance:
```

Comment:
Here you've reimplemented a slightly different version of `np.allclose`.  You might just want to use it instead.
```
if np.allclose((input_xtilt, input_ytilt), (bkg_xtilt, bkg_ytilt), atol=1e-8, rtol=0):
```
The parameters `atol` and `rtol` are important for different sized numbers, so I'm not sure what those should be.  Numbers very close to zero need special consideration.  Do we know what range these GWA tilt numbers are?